### DISCUSSION TITLE
middleware causes exceptions to not be raised/handled silently

### DISCUSSION THREAD

**Discussion by fraser-langton** [encode/starlette#1976] [Potential Issue] [answered]
From [issue in FastAPI](https://github.com/tiangolo/fastapi/issues/5173) - issue was said to be from starlette

Was initially noticed in FastAPI 0.74.0, was fixed after 0.79.0 but has since regressed (FastAPI 0.88.0 it isn't working)

When exceptions are raised on a subapp, the exceptions are not propagated all the way to see in console
In the example code
/info raises an exception and the full stacktrace is seen in console
/private/info does not raise the exception and only `INFO: 127.0.0.1:56308 - "GET /info HTTP/1.1" 500 Internal Server Error` is shown in console

```python
import uvicorn
from fastapi import FastAPI
from starlette.middleware.base import BaseHTTPMiddleware


app = FastAPI()


@app.get("/info")
def info():
    # raises Exception as expected, the traceback is seen in console
    raise Exception


private_api = FastAPI()


@private_api.get("/info")
def info():
    # exception is handled silently, no traceback is seen in console
    raise Exception


app.mount("/private", private_api)


class Middleware(BaseHTTPMiddleware):

    async def dispatch(self, request, call_next):
        return await call_next(request)


app.add_middleware(Middleware)  # when this is removed, the exceptions are raised for all routes


if __name__ == '__main__':
    uvicorn.run(app, port=8000)
```

**Comment by Kludex** [comment, marked as answer]
Created an issue for this: https://github.com/encode/starlette/issues/1977.

**Reply by Kludex** [reply]
Solved on https://github.com/encode/starlette/pull/1940.

**Comment by jgaul3** [comment]
Hello!

I was tracking down why we couldn't bump our FastAPI version and found this discussion and fix. Exactly our issue! I was thrilled. The trouble is that the fix got un-fixed [here](https://github.com/encode/starlette/pull/2194)... I did my best to look through the comments and it seems like the consensus is that Middleware handling is a headache at the best of times and subapps make it even harder.

Not sure what the intended behavior is but wanted to see if we could re-open this discussion. Thanks for your time!

**Reply by Nnonexistent** [reply]
Seems to be fixed back at v0.46

**Reply by Kludex** [reply]
What does "fixed back" means?

**Reply by Nnonexistent** [reply]
As I understood the bug was initially solved in #1940. Then it breaks again at some point. I have this bug reproduced in `v0.45.3` but not in `v0.46`.

**Reply by Kludex** [reply]
So you mean it's all good or not?

**Reply by Nnonexistent** [reply]
Yes, all good now! :)
